### ISSUE TITLE
Is there a current method to create a subscription id?

### ISSUE THREAD

**Issue by ebenoist** [go-stomp/stomp#4]
I need to guarantee a consistent and unique subscription id across a number of processes, is there a method by which to set the id with the current library when subscribing? I see that a random number is generated, but it doesn't seem if there is a way to override it.

Thank you!

**Comment by jjeffery** [comment]
The subscription id is set in the allocateId function. The value generated is not random, it is an incrementing value. There is no way to set the subscription id without modifying this function.

I wonder why you need to do this. The subscription id is just an arbitrary identifier used to identify multiple subscriptions on the same connection to the server. Is your requirement something server specific?

**Comment by ebenoist** [comment]
It likely is server specific, but I might also be missing something. With the hornetq cluster I'm working with a unique subscription id will ensure that multiple clients on separate hosts will receive the same message (each process acks the message for that subscription id). In addition subscription ids can be reused to create a durable topic. 

I notice that the [ruby stomp gem](https://github.com/stompgem/stomp/blob/6b2c0f6b4fcb0ec6d25f84b67143ab8b730ddfde/lib/client/utils.rb) allows an id to be sent in via the headers, but if it isn't present the id is generated using a hashing function. Would you be up for a PR that does something similar? This way a client can optionally send in a subscription id, but in its absence, the existing code could call back the the `allocateId` function.

**Comment by ebenoist** [comment]
The following is working well for me. I noticed that a similar function existed before this code was forked into this repo.

```Go
// Subscribe creates a subscription on the STOMP server.
// The subscription has a destination, and messages sent to that destination
// will be received by this subscription. A subscription has a channel
// on which the calling program can receive messages.
func (c *Conn) Subscribe(destination string, ack AckMode) (*Subscription, error) {
	return c.SubscribeWithHeaders(destination, ack, NewHeader())
}

func (c *Conn) SubscribeWithHeaders(destination string, ack AckMode, userDefined *Header) (*Subscription, error) {
	ch := make(chan *Frame)
	headers := userDefined.Clone()

	if _, ok := headers.Contains(frame.Id); !ok {
		headers.Add(frame.Id, allocateId())
	}

	headers.Add(frame.Destination, destination)
	headers.Add(frame.Ack, ack.String())

	f := NewFrame(frame.SUBSCRIBE)
	f.Header = headers

	request := writeRequest{Frame: f, C: ch}

	sub := &Subscription{
		id:          headers.Get(frame.Id),
		destination: destination,
		conn:        c,
		ackMode:     ack,
		C:           make(chan *Message, 16),
	}
	go sub.readLoop(ch)

	c.writeCh <- request
	return sub, nil
}
```

**Comment by ebenoist** [comment]
This was addressed with my pr #7 .
